drivers/serial/serial_mxc.c

```diff
@@ -318,10 +329,36 @@ static int mxc_serial_pending(struct udevice *dev, bool input)
 		return sr2 & USR2_TXDC ? 0 : 1;
 }
 
+static int barebox_mxc_serial_pending(struct udevice *dev, bool input)
+{
+	struct mxc_serial_plat *plat = dev_get_plat(dev);
+	struct mxc_uart *const uart = plat->reg;
+	uint32_t ts = readl(&uart->ts);
+
+	if (input)
+            return ts & UTS_RXEMPTY ? 0 : 1;
+	else
+            return ts & UTS_TXEMPTY ? 0 : 1;
+}
+static int barebox_mxc_serial_getc(struct udevice *dev)
+{
+	struct mxc_serial_plat *plat = dev_get_plat(dev);
+	struct mxc_uart *const uart = plat->reg;
+	while (readl(&uart->ts) & UTS_RXEMPTY);
+	return readl(&uart->rxd) & URXD_RX_DATA;
+}
+static int barebox_mxc_serial_putc(struct udevice *dev, const char ch)
+{
+	struct mxc_serial_plat *plat = dev_get_plat(dev);
+	struct mxc_uart *const uart = plat->reg;
+        while(readl(&uart->ts) & UTS_TXFULL);
+        writel(ch, &uart->txd);
+        return 0;
+}
 static const struct dm_serial_ops mxc_serial_ops = {
-	.putc = mxc_serial_putc,
-	.pending = mxc_serial_pending,
-	.getc = mxc_serial_getc,
+	.putc = barebox_mxc_serial_putc,
+	.pending = barebox_mxc_serial_pending,
```

Comment:
I believe the difference between RDR in UARTx_USR2 and RXEMPTY in UARTx_UTS is:
- "RxFIFO Empty. Indicates the RxFIFO is empty."
  Okay, RX fifo is empty - without any precondition...
- "Receive Data Ready-Indicates that at least 1 character is received and written to the RxFIFO. If the
URXD register is read and there is only 1 character in the RxFIFO, RDR is automatically cleared."
  We need to receive at least one character, which must not be read, so that this flag is set.

From the function name `pending` I would say the RDR makes more sense.

For TX both could be used:
- "TxFIFO Empty. Indicates that the TxFIFO is empty."
- "Transmitter Complete. Indicates that the transmit buffer (TxFIFO) and Shift Register is empty; therefore
the transmission is complete. TXDC is cleared automatically when data is written to the TxFIFO."

Reply:
thnx; - after comparing both implementations again - with your input in mind - i think i found the issue with the u-boots implementation: the putc path needs to make use of the TXFIFO, otherwise  the message gets "corrupted" = does not make it to the MCU, and therefore no response is created...
-> see follow-up patch